### PR TITLE
[W5.7][T16-4] Gabriel Yik

### PR DESCRIPTION
**Pull request by GabrielYik**
Add new edit command.

Some commits have been squashed when a soft revert was used to remove an unwanted commit.

### CODE REVIEW

**Comment by Hilda-Ang** [comment]
Great job on implementing the edit command incrementally. Would suggest to update I/O tests (update input.txt and expected.txt) besides the JUnit tests. Also, have you added the edit command to the command list in help (HelpCommand.java)?

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 1]
```diff
@@ -1,11 +1,12 @@
 package seedu.addressbook.common;
```
Comment (rongjiecomputer) [inline test/java/seedu/addressbook/common/UtilsTest.java]: It looks like unrelated commit for another PR is getting added into this PR. Can you remove it?

Comment (GabrielYik) [reply test/java/seedu/addressbook/common/UtilsTest.java]: I'm not sure how to. It was not in the feature/edit branch. It's supposed to be in the main branch

Comment (rongjiecomputer) [reply test/java/seedu/addressbook/common/UtilsTest.java]: You can do so by running `git revert a77790080dd242233f6f2aa873ac7c00c981ef5a` to revert the commit.

File: src/seedu/addressbook/commands/EditCommand.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand(String name,
+                      String phone, boolean isPhonePrivate,
+                      String email, boolean isEmailPrivate,
+                      String address, boolean isAddressPrivate,
+                      Set<String> tags) throws IllegalValueException {
+        final Set<Tag> tagSet = new HashSet<>();
```
Comment (sushinoya) [inline src/seedu/addressbook/commands/EditCommand.java:44]: Why is this set `final`? I get that nothing else in the function should modify it but your code ensures that. And nothing outside the function has access to it since it is a local variable.

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -38,6 +38,35 @@ public void addPerson(Person toAdd) throws DuplicatePersonException {
...
+     * It silences {@code DuplicatePersonException} since this method is
+     * called after {@code PersonNotFoundException} is thrown.
+     * Both exceptions are mutually exclusive.
+     *
+     */
+    private void addPersonSafe(Person toAdd) {
+        try {
+            addPerson(toAdd);
+        } catch(DuplicatePersonException dpe) {
+            // Will not be run
```
Comment (sushinoya) [inline src/seedu/addressbook/data/AddressBook.java:53]: I think it is not safe to assume that just because the exception will not be raised now means that no changes to the project might every raise the exception. I think it should be handled well still.

**Review by sushinoya — COMMENTED** [review]
Good job overall!

For commit messages, it is possible that one commit contains changes to multiple files. Do you intend to have the names of all the files changed in your commit messages?

Close the PR after reading my comments :)
